.github/workflows/ci.yml

```
@@ -0,0 +1,40 @@
+name: CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build-with-various-java-version:
```

```suggestion
  build-with-various-java-versions:
```
It's a minor English thing, but it might be better to `versions` instead of `version` here.